browser/fragments/platform-browser.ts

```diff
@@ -8,36 +12,37 @@ function allowComet() {
 	/* xhr requests from local files are unreliable in some browsers, such as Chrome 65 and higher -- see eg
 	 * https://stackoverflow.com/questions/49256429/chrome-65-unable-to-make-post-requests-from-local-files-to-flask
 	 * So if websockets are supported, then just forget about comet transports and use that */
-	var loc = global.location;
+	const loc = global.location;
 	return (!global.WebSocket || !loc || !loc.origin || loc.origin.indexOf("http") > -1);
 }
 
-var userAgent = global.navigator && global.navigator.userAgent.toString();
-var currentUrl = global.location && global.location.href;
+const userAgent = global.navigator && global.navigator.userAgent.toString();
+const currentUrl = global.location && global.location.href;
 
-var Platform = {
-        agent: 'browser',
+const Platform: IPlatform = {
+	agent: 'browser',
 	logTimestamps: true,
 	userAgent: userAgent,
 	currentUrl: currentUrl,
-	noUpgrade: userAgent && userAgent.match(/MSIE\s8\.0/),
+	noUpgrade: userAgent && !!userAgent.match(/MSIE\s8\.0/),
 	binaryType: 'arraybuffer',
-	WebSocket: global.WebSocket || global.MozWebSocket,
+	WebSocket: global.WebSocket || MozWebSocket,
 	xhrSupported: global.XMLHttpRequest && 'withCredentials' in new XMLHttpRequest(),
 	jsonpSupported: typeof(document) !== 'undefined',
 	allowComet: allowComet(),
 	streamingSupported: true,
 	useProtocolHeartbeats: true,
 	createHmac: null,
-	msgpack: msgpack,
+	// TODO: type this properly
+	msgpack: msgpack as unknown as typeof import('@ably/msgpack-js'),
 	supportsBinary: !!global.TextDecoder,
 	preferBinary: false,
 	ArrayBuffer: global.ArrayBuffer,
 	atob: global.atob,
-	nextTick: typeof global.setImmediate !== 'undefined' ? global.setImmediate.bind(global) : function(f) { setTimeout(f, 0); },
-	addEventListener: global.addEventListener,
+	nextTick: typeof global.setImmediate !== 'undefined' ? global.setImmediate.bind(global) : function(f: () => void) { setTimeout(f, 0); },
+	addEventListener: global.addEventListener as unknown as null, // This is not ideal but TypeScript doesn't seem to allow anything else here
```

Comment:
> `global.addEventListener as unknown as null, // This is not ideal but TypeScript doesn't seem to allow anything else here`

removing `as unknown as null` it seems to compile fine for me?

Reply:
Yeah you're right, I've removed it now